internal/services/monitor/monitor_diagnostic_setting_resource.go

```diff
@@ -498,8 +553,15 @@ func resourceMonitorDiagnosticSettingRead(d *pluginsdk.ResourceData, meta interf
 				return fmt.Errorf("setting `enabled_log`: %+v", err)
 			}
 
-			if err := d.Set("metric", flattenMonitorDiagnosticMetrics(resp.Model.Properties.Metrics)); err != nil {
-				return fmt.Errorf("setting `metric`: %+v", err)
+			enabledMetrics := flattenMonitorDiagnosticEnabledMetrics(resp.Model.Properties.Metrics)
+			if err = d.Set("enabled_metric", enabledMetrics); err != nil {
```

Comment:
Because we don't use `enabledMetrics` anywhere else, we  don't need to declare a variable for it

```suggestion
			if err = d.Set("enabled_metric", flattenMonitorDiagnosticEnabledMetrics(resp.Model.Properties.Metrics)); err != nil {
```